assets/js/modules/search-console/components/dashboard/DashboardPopularKeywordsWidget.js

```diff
@@ -20,72 +20,48 @@
-import { numFmt, untrailingslashit } from '../../../../util';
-import { getDataTableFromData } from '../../../../components/data-table';
 import whenActive from '../../../../util/when-active';
 import PreviewTable from '../../../../components/PreviewTable';
 import SourceLink from '../../../../components/SourceLink';
 import { isZeroReport } from '../../util';
 import TableOverflowContainer from '../../../../components/TableOverflowContainer';
 import { generateDateRangeArgs } from '../../util/report-date-range-args';
+import ReportTable from '../../../../components/ReportTable';
+import Link from '../../../../components/Link';
+import Decimal from '../../../../components/Num/Decimal';
 const { useSelect } = Data;
 
 function DashboardPopularKeywordsWidget( { Widget, WidgetReportZero, WidgetReportError } ) {
-	const {
-		data,
-		error,
-		loading,
-		baseServiceURL,
-	} = useSelect( ( select ) => {
-		const store = select( STORE_NAME );
-		const domain = store.getPropertyID();
-
-		const { startDate, endDate } = select( CORE_USER ).getDateRangeDates( { offsetDays: DATE_RANGE_OFFSET } );
-		const args = {
-			startDate,
-			endDate,
-			dimensions: 'query',
-			limit: 10,
-		};
-
-		const baseServiceURLArgs = {
-			resource_id: domain,
-			...generateDateRangeArgs( { startDate, endDate } ),
-		};
-
-		const url = select( CORE_SITE ).getCurrentEntityURL();
-		const isDomainProperty = select( STORE_NAME ).isDomainProperty();
-		const referenceSiteURL = untrailingslashit( select( CORE_SITE ).getReferenceSiteURL() );
-		if ( url ) {
-			args.url = url;
-			baseServiceURLArgs.page = `!${ url }`;
-		} else if ( isDomainProperty && referenceSiteURL ) {
-			baseServiceURLArgs.page = `*${ referenceSiteURL }`;
-		}
-
-		return {
-			data: store.getReport( args ),
-			error: store.getErrorForSelector( 'getReport', [ args ] ),
-			loading: ! store.hasFinishedResolution( 'getReport', [ args ] ),
-			baseServiceURL: store.getServiceURL( {
-				path: '/performance/search-analytics',
-				query: baseServiceURLArgs,
-			} ),
-		};
-	} );
+	const url = useSelect( ( select ) => select( CORE_SITE ).getCurrentEntityURL() );
+	const dateRangeDates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( { offsetDays: DATE_RANGE_OFFSET } ) );
+	const { startDate, endDate } = dateRangeDates;
+	const reportArgs = {
+		startDate,
+		endDate,
+		dimensions: 'query',
+		limit: 10,
+		url: url || undefined,
+	};
+	const data = useSelect( ( select ) => select( STORE_NAME ).getReport( reportArgs ) );
+	const error = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ reportArgs ] ) );
+	const loading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ reportArgs ] ) );
+	const baseServiceURL = useSelect( ( select ) => select( STORE_NAME ).getServiceReportURL( {
+		...generateDateRangeArgs( dateRangeDates ),
+		page: url ? `!${ url }` : undefined,
+	} ) );
```

Comment:
There's a regression here, see above: For a domain property this should, if no entity URL exists, still ensure that the site URL is included as `page` query parameter, to filter it down as needed; in the new code the `page` query parameter is instead omitted.

Was there a need to implement a `getServiceReportURL` selector here? It's probably a good idea to have it, but feels a bit out of scope in this already large PR. I guess since this is the only usage here, we should be okay keeping it in.

Reply:
> Was there a need to implement a `getServiceReportURL` selector here? It's probably a good idea to have it, but feels a bit out of scope in this already large PR.

It wasn't strictly required, no, but it helped reduce a substantial amount of duplication as well as handle the domain property logic which is something that individual components shouldn't need to be concerned with. The effort was fairly minimal so I think it was a worthwhile enhancement to include.